src/pages/ServiceDetails/ServiceMetrics.tsx

```diff
@@ -37,66 +38,77 @@ class ServiceMetrics extends React.Component<ServiceId, ServiceMetricsState> {
   }
 
-  onOptionsChanged = (options: MetricsOptions) => this.fetchMetrics(options);
+  onOptionsChanged = (options: MetricsOptions) => {
+    if (this.state.pollMetrics !== undefined) {
+      clearTimeout(this.state.pollMetrics);
+    }
+    this.fetchMetrics(options, this);
+  };
 
-  fetchMetrics(options: MetricsOptions) {
+  fetchMetrics(options: MetricsOptions, object: any) {
     options.filters = ['request_count', 'request_size', 'request_duration', 'response_size'];
-    this.setState({ loading: true });
-    API.getServiceMetrics(this.props.namespace, this.props.service, options)
+    object.setState({ loading: true });
+    API.getServiceMetrics(object.props.namespace, object.props.service, options)
       .then(response => {
         const metrics: M.Metrics = response['data'];
-        this.setState({
+        object.setState({
           loading: false,
-          requestCountIn: this.nameMetric(
+          requestCountIn: object.nameMetric(
             metrics.metrics['request_count_in'],
             'Request volume (ops)',
             options.byLabelsIn
           ),
-          requestCountOut: this.nameMetric(
+          requestCountOut: object.nameMetric(
             metrics.metrics['request_count_out'],
             'Request volume (ops)',
             options.byLabelsOut
           ),
-          requestSizeIn: this.nameHistogram(
+          requestSizeIn: object.nameHistogram(
             metrics.histograms['request_size_in'],
             'Request size (bytes)',
             options.byLabelsIn
           ),
-          requestSizeOut: this.nameHistogram(
+          requestSizeOut: object.nameHistogram(
             metrics.histograms['request_size_out'],
             'Request size (bytes)',
             options.byLabelsOut
           ),
-          requestDurationIn: this.nameHistogram(
+          requestDurationIn: object.nameHistogram(
             metrics.histograms['request_duration_in'],
             'Request duration (seconds)',
             options.byLabelsIn
           ),
-          requestDurationOut: this.nameHistogram(
+          requestDurationOut: object.nameHistogram(
             metrics.histograms['request_duration_out'],
             'Request duration (seconds)',
             options.byLabelsOut
           ),
-          responseSizeIn: this.nameHistogram(
+          responseSizeIn: object.nameHistogram(
             metrics.histograms['response_size_in'],
             'Response size (bytes)',
             options.byLabelsIn
           ),
-          responseSizeOut: this.nameHistogram(
+          responseSizeOut: object.nameHistogram(
             metrics.histograms['response_size_out'],
             'Response size (bytes)',
             options.byLabelsOut
           )
         });
+        if (options.pollInterval !== undefined) {
+          if (options.pollInterval > 0) {
+            object.setState({ pollMetrics: setTimeout(object.fetchMetrics, options.pollInterval, options, object) });
```

Comment:
`setInterval` is usually better suited for repeated tasks.
Also, the state of `pollInterval` shouldn't necessarily changes on `fetchMetrics`. So maybe it's better to set `pollInterval` from `onOptionsChanged` and clear and/or set interval from there and not from `fetchMetrics`